cli/command/stack/kubernetes/cli.go

```diff
@@ -57,7 +53,19 @@ func WrapCli(dockerCli command.Cli, opts Options) (*KubeCli, error) {
 			kubeConfig = filepath.Join(homedir.Get(), ".kube/config")
 		}
 	}
-	config, err := kubernetes.NewKubernetesConfig(kubeConfig)
+
+	clientConfig := kubernetes.NewKubernetesConfig(kubeConfig)
+
+	configNamespace, _, err := clientConfig.Namespace()
+	if err != nil {
+		return nil, err
```

Comment:
Looks like this code is executed (possibly showing an error) even if the namespace is overridden through `opts.Namespace`. Retrieving the namespace from the config should probably be done conditional (i.e., if there's no `--namespace` option set)?

Reply:
Good point, I'll invert the condition below, but the code in here is likely going to change when I address your next comment about ```--namespace```.